collector/src/main/java/io/prometheus/jmx/JmxScraper.java

```diff
@@ -301,29 +296,62 @@ public void recordBean(
             String attrType,
             String attrDescription,
             Object value) {
-            System.out.println(domain +
-                               beanProperties + 
-                               attrKeys +
-                               attrName +
-                               ": " + value);
+            logger.log(Level.INFO, domain +
+                    beanProperties +
+                    attrKeys +
+                    attrName +
+                    ": " + value);
+        }
+    }
+
+    private static class DumbFormatter extends SimpleFormatter {
+        @Override
+        public synchronized String format(LogRecord record) {
+            return record.getMessage() + "\r\n";
         }
     }
 
     /**
      * Convenience function to run standalone.
      */
     public static void main(String[] args) throws Exception {
-      List<ObjectName> objectNames = new LinkedList<ObjectName>();
-      objectNames.add(null);
-      if (args.length >= 3){
-            new JmxScraper(args[0], args[1], args[2], false, objectNames, new LinkedList<ObjectName>(), new StdoutWriter()).doScrape();
+        // Clean existing log handlers
+        for (Handler h : logger.getHandlers()) {
+            logger.removeHandler(h);
+        }
+        for (Handler h : LoggerJMX.getHandlers()) {
+            LoggerJMX.removeHandler(h);
+        }
+        for (Handler h : Logger.getLogger("").getHandlers()) {
+            Logger.getLogger("").removeHandler(h);
+        }
+
+        // Create our stdout handler
+        SimpleFormatter fmt = new DumbFormatter();
+        StreamHandler sh = new StreamHandler(System.out, fmt);
+
+
+        Set<String> argss = new HashSet<String>(args.length);
+        argss.addAll(Arrays.asList(args));
+        if (argss.contains("--time_jmx_call")) {
```

Comment:
Well It is kind of troublesome to do it the other way around and just use positionnal arguments.
Even if it is a bit overkill, it is the easiest way.